src/js/view/controls/components/menu/tizen-menu.js

```diff
@@ -0,0 +1,232 @@
+
+        this.el.addEventListener('animationend', this.onTransition);
+    }
+
+    setupMenu(menuName, menuTitle, items, onItemSelect, defaultItemIndex, itemOptions) {
+        if (!items || items.length <= 1) {
+            this.removeMenu(menuName);
+            return;
+        }
+        let menu = this.children[menuName];
+    
+    
+        if (!menu) {
+            // We pass null as parent menu to avoid using the standard menu HTML heirarchy.
+            menu = new Menu(menuName, menuTitle, null, this.localization, TizenSubmenuTemplate);
+            menu.parentMenu = this;
+            menu.itemsContainer = new UI(menu.el.querySelector('.jw-settings-submenu-items'));
+            const noop = () => {};
+            // replace open, close and toggle with noops. Submenus always visible in Tizen.
+            menu.open = menu.close = menu.toggle = noop;
+            menu.el.classList.add('jw-settings-submenu-active');
+            // add tizen control listener
+            this.appendMenu(menu);
+        }
+
+        menu.setMenuItems(
+            menu.createItems(items, onItemSelect, itemOptions), 
+            defaultItemIndex
+        );
+    }
+    
+    onInteraction(evt) {
+        const onNavigateMenu = function(isUp) {
+            const next = isUp ? previousSibling(evt.target) : nextSibling(evt.target);
+            if (next) {
+                next.focus({ preventScroll: true });
+            } else {
+                const nextMenu = isUp ? this.children.captions : this.children.audioTracks;
+                if (nextMenu && !nextMenu.el.contains(evt.target)) {
+                    const items = nextMenu.items;
+                    (isUp ? items[items.length - 1] : items[0]).el.focus({ preventScroll: true });
+                }
+            }
+            this.trigger(USER_ACTION);
+        };
+
+        switch (evt.keyCode) {
+            case 38: // Up
+                onNavigateMenu.call(this, true);
+                break;
+            case 40: // Down
+                onNavigateMenu.call(this, false);
+                break;
+            case 37: // Left
+                this.close();
+                if (this.controlbar) {
+                    this.controlbar.elements.settingsButton.element().focus();
+                }
+                break;
+            case 415: // Play
+            case 10252: // Play/Pause
+                this.close(evt, true);
+                this.api.play({ reason: 'settingsInteraction' });
+                break;
+            case 10009: // Back
+                this.close(evt);
+                break;
+            default:
+                break;
+        }
+    }
+
+    onVisibility(evt) {
+        const el = this.el;
+        
+        if (evt.visible) {
+            this.api.pause({ reason: 'settingsInteraction' });
+            document.addEventListener('keydown', this.onInteraction);
+            el.classList.remove('jw-settings-transition-close');
```

Comment:
`el.classList` is referenced multiple times in both conditional blocks. We should use a const